ludwig/encoders/text_encoders.py

```diff
@@ -82,30 +97,45 @@ def __init__(
             pretrained_model_name_or_path
         )
         self.reduce_output = reduce_output
+        self.transformer.trainable = False
+
 
     def call(self, inputs, training=None, mask=None):
-        transformer_outputs = self.transformer(inputs, training=training)
+        transformer_outputs = self.transformer(
+            inputs, 
+            training=training,
+            attention_mask=mask,
+            token_type_ids=tf.zeros_like(inputs)
+        )
         hidden = transformer_outputs[0]
         hidden = reduce_sequence(hidden, self.reduce_output)
-        return hidden
+        return {'encoder_output': hidden}
 
 
 class TransformerXLEncoder(Layer):
 
     def __init__(
             self,
+            pretrained_model_name_or_path='transfo-xl-wt103',
             reduce_output='sum',
             **kwargs
     ):
         super(TransformerXLEncoder, self).__init__()
-        self.transformer = TFTransfoXLModel.from_pretrained('transfo-xl-wt103')
+        self.transformer = TFTransfoXLModel.from_pretrained(
+            pretrained_model_name_or_path
+        )
         self.reduce_output = reduce_output
+        self.transformer.trainable = False
+
 
     def call(self, inputs, training=None, mask=None):
-        transformer_outputs = self.transformer(inputs, training=training)
+        transformer_outputs = self.transformer(
```

Comment:
this call looks different from the other encoders, specifically attention_mask and token_type_ids are missing. Is there a specific reason?

Reply:
Yup. This encoder doesn't have the attention_mask and token_type_ids as parameters.